pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -49,44 +50,104 @@ const (
+	pluginConfig := pluginNameToConfig(args)
 	for name, factory := range r {
-		// TODO: 'nil' should be replaced by plugin config.
-		p, err := factory(nil, f)
+		// initialize only needed plugins
+		if _, ok := pg[name]; !ok {
+			continue
+		}
+
+		// find the config args of a plugin
+		pc := pluginConfig[name]
+
+		p, err := factory(pc, f)
 		if err != nil {
 			return nil, fmt.Errorf("error initializing plugin %v: %v", name, err)
 		}
 		f.plugins[name] = p
+	}
 
-		// TODO: For now, we assume any plugins that implements an extension
-		// point wants to be called at that extension point. We should change this
-		// later and add these plugins based on the configuration.
-		if qsp, ok := p.(QueueSortPlugin); ok {
-			f.queueSortPlugins = append(f.queueSortPlugins, qsp)
+	for _, r := range plugins.EnabledPlugins.Reserve {
+		if pg, ok := f.plugins[r.Name]; ok {
+			p, ok := pg.(ReservePlugin)
+			if !ok {
+				return nil, fmt.Errorf("plugin %v does not extend reserve plugin", r.Name)
+			}
+			f.reservePlugins = append(f.reservePlugins, p)
+		} else {
+			return nil, fmt.Errorf("reserve plugin %v does not exist", r.Name)
 		}
+	}
 
-		if rp, ok := p.(ReservePlugin); ok {
-			f.reservePlugins = append(f.reservePlugins, rp)
+	for _, pb := range plugins.EnabledPlugins.PreBind {
+		if pg, ok := f.plugins[pb.Name]; ok {
+			p, ok := pg.(PrebindPlugin)
+			if !ok {
+				return nil, fmt.Errorf("plugin %v does not extend prebind plugin", pb.Name)
+			}
+			f.prebindPlugins = append(f.prebindPlugins, p)
+		} else {
+			return nil, fmt.Errorf("prebind plugin %v does not exist", pb.Name)
 		}
-		if pp, ok := p.(PrebindPlugin); ok {
-			f.prebindPlugins = append(f.prebindPlugins, pp)
+	}
+
+	for _, ur := range plugins.EnabledPlugins.Unreserve {
+		if pg, ok := f.plugins[ur.Name]; ok {
+			p, ok := pg.(UnreservePlugin)
+			if !ok {
+				return nil, fmt.Errorf("plugin %v does not extend unreserve plugin", ur.Name)
+			}
+			f.unreservePlugins = append(f.unreservePlugins, p)
+		} else {
+			return nil, fmt.Errorf("unreserve plugin %v does not exist", ur.Name)
 		}
-		if up, ok := p.(UnreservePlugin); ok {
-			f.unreservePlugins = append(f.unreservePlugins, up)
+	}
+
+	for _, pr := range plugins.EnabledPlugins.Permit {
+		if pg, ok := f.plugins[pr.Name]; ok {
+			p, ok := pg.(PermitPlugin)
+			if !ok {
+				return nil, fmt.Errorf("plugin %v does not extend permit plugin", pr.Name)
+			}
+			f.permitPlugins = append(f.permitPlugins, p)
+		} else {
+			return nil, fmt.Errorf("permit plugin %v does not exist", pr.Name)
 		}
-		if pr, ok := p.(PermitPlugin); ok {
-			f.permitPlugins = append(f.permitPlugins, pr)
+	}
+
+	for _, qs := range plugins.EnabledPlugins.QueueSort {
+		if pg, ok := f.plugins[qs.Name]; ok {
+			p, ok := pg.(QueueSortPlugin)
+			if !ok {
+				return nil, fmt.Errorf("plugin %v does not extend queue sort plugin", qs.Name)
+			}
+			f.queueSortPlugins = append(f.queueSortPlugins, p)
+
+			if len(f.queueSortPlugins) >= 2 {
```

Comment:
nit: Please change `>= 2` to `> 1` and remove the blank line before the if statement.